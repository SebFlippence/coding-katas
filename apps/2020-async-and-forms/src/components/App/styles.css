html, body {
  height: 100%;
  margin: 0;
  font-family: sans-serif;
}

#root {
  height: 100%;
}

.exercise-list__item {
  display: block;
}
