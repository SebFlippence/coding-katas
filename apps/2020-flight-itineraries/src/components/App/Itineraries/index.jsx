import React from 'react';
import BpkCard from 'bpk-component-card';
import BpkButton from 'bpk-component-button';
import {
  BpkGridContainer,
  BpkGridRow,
  BpkGridColumn,
} from 'bpk-component-grid';

import STYLES from './index.scss';
import Legs from './Legs';

const Itinerary = ({ item }) => (
  <BpkGridContainer fullWidth>
    <BpkGridRow>
      <BpkGridColumn width={8}>
        <span className={STYLES.price}>{item.price}</span>
        <br />
        <span className={STYLES.agent}>{item.agent}</span>
      </BpkGridColumn>
      <BpkGridColumn width={1}>
        <BpkButton>Select</BpkButton>
      </BpkGridColumn>
    </BpkGridRow>
  </BpkGridContainer>
);

const Itineraries = ({ flights }) =>
  flights.itineraries.map(itinerary => (
    <section key={itinerary.id} className={STYLES.item}>
      <BpkCard className={STYLES.card}>
        <Legs
          className="Legs"
          data={flights.legs.filter(leg => itinerary.legs.includes(leg.id))}
        />
        <Itinerary item={itinerary} />
      </BpkCard>
    </section>
  ));

export default Itineraries;
