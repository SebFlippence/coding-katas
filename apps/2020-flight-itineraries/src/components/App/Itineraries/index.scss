.card {
  width: 365px;
  margin-left: 10px;
}

.item {
  margin-bottom: 20px;
}

.price {
  font-size: 1.5rem;
}

.agent {
  color: #ccc;
}
