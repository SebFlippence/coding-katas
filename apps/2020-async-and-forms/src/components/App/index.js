import React from "react";

import Complexity from "../Complexity";
import Form from "../Form";

import "./styles.css";

export default () => {
  const page = window.location.pathname.slice(1);

  return (
    <>
      {page.length === 0 && (
        <ul className="exercise-list">
          <li className="exercise-list__item">
            <a href="/complexity">Complexity</a>
          </li>
          <li className="exercise-list__item">
            <a href="/form">Form</a>
          </li>
        </ul>
      )}

      {page === "complexity" && <Complexity />}

      {page === "form" && <Form />}
    </>
  );
};
