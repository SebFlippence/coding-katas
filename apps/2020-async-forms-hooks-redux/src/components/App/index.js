import React from 'react';

import Complexity from '../Complexity';
import Form from '../Form';
import FormHooks from '../FormHooks';
import CustomHook from '../CustomHook';
import ReduxToolkit from '../ReduxToolkit';

import './styles.css';

export default () => {
  const page = window.location.pathname.slice(1);

  return (
    <>
      {page.length === 0 && (
        <ul className="exercise-list">
          <li className="exercise-list__item">
            <a href="/complexity">Complexity</a>
          </li>
          <li className="exercise-list__item">
            <a href="/form">Form</a>
          </li>
          <li className="exercise-list__item">
            <a href="/form-hooks">React Form Hook</a>
          </li>
          <li className="exercise-list__item">
            <a href="/custom-hook">Custom Hook</a>
          </li>
          <li className="exercise-list__item">
            <a href="/redux-toolkit">Redux Toolkit</a>
          </li>
        </ul>
      )}

      {page === 'form' && <Form />}

      {page === 'form-hooks' && <FormHooks />}

      {page === 'custom-hook' && <CustomHook />}

      <div style={{ margin: 10 }}>
        {page === 'complexity' && <Complexity />}

        {page === 'redux-toolkit' && <ReduxToolkit />}
      </div>
    </>
  );
};
