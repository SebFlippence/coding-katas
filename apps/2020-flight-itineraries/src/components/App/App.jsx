import React, { useEffect, useState } from 'react';
import BpkText from 'bpk-component-text';

import Header from '../Header';

import STYLES from './App.scss';
import Itineraries from './Itineraries';

const getClassName = className => STYLES[className] || 'UNKNOWN';

const App = () => {
  const [flights, setFlights] = useState();

  useEffect(() => {
    async function fetchData() {
      const response = await fetch('flights.json');
      if (!response.ok) {
        throw new Error('There was a problem loading flights');
      }
      setFlights(await response.json());
    }
    fetchData();
  }, []);

  return (
    <div className={getClassName('App')}>
      <Header />
      <main className={getClassName('App__main')}>
        {flights ? (
          <Itineraries flights={flights} />
        ) : (
          <BpkText tagName="p">Looking for flights...</BpkText>
        )}
      </main>
    </div>
  );
};

export default App;
