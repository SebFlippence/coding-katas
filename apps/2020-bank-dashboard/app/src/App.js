import React, { useEffect } from 'react';
import { observer, inject } from 'mobx-react';
import './App.css';

function App({ bankStore }) {
  useEffect(() => {
    bankStore.getTransactions();
  }, [bankStore]);

  return (
    <div className="App">
      <h1>Transactions</h1>
      {!bankStore.transactions.length ? (
        <p>Loading...</p>
      ) : (
        <table>
          <thead>
            <tr>
              <th>Amount</th>
              <th>User</th>
              <th>Retailer</th>
            </tr>
          </thead>
          <tbody>
            {bankStore.transactions.map(transaction => (
              <tr key={transaction.id}>
                <td>
                  {transaction.amount} {transaction.currency}
                </td>
                <td>{bankStore.users[transaction.userId] ? bankStore.users[transaction.userId].name : 'Loading...'}</td>
                <td>
                  {bankStore.retailers[transaction.retailerId]
                    ? `${bankStore.retailers[transaction.retailerId].name} ${
                        bankStore.retailers[transaction.retailerId].city
                      }`
                    : 'Loading...'}
                </td>
              </tr>
            ))}
          </tbody>
        </table>
      )}
    </div>
  );
}

export default inject(({ bankStore }) => ({ bankStore }))(observer(App));
