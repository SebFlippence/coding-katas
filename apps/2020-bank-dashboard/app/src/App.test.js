import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'mobx-react';
import App from './App';

const renderWithProviderStore = (ui, store) => render(<Provider {...store}>{ui}</Provider>);

describe('App', () => {
  let mockStore;
  beforeEach(() => {
    mockStore = {
      transactions: [],
      users: {},
      retailers: {},
      getTransactions: jest.fn(),
    };
  });

  test('renders loading screen', () => {
    const { container } = renderWithProviderStore(<App />, { bankStore: mockStore });
    expect(mockStore.getTransactions).toHaveBeenCalledTimes(1);
    expect(container).toMatchSnapshot();
  });

  test('renders with transaction data with other data loading', () => {
    const bankStore = { ...mockStore, transactions: [{ id: 1, amount: '1', currency: 'USD' }] };

    const { container } = renderWithProviderStore(<App />, { bankStore });
    expect(container).toMatchSnapshot();
  });

  test('renders with transaction & user data, with retailer data loading', () => {
    const bankStore = {
      ...mockStore,
      transactions: [{ id: 1, amount: '1', currency: 'USD', userId: 1 }],
      users: { '1': { name: 'Joe Bloggs' } },
    };

    const { container } = renderWithProviderStore(<App />, { bankStore });
    expect(container).toMatchSnapshot();
  });

  test('renders with transaction, user & retailer data', () => {
    const bankStore = {
      ...mockStore,
      transactions: [
        { id: 1, amount: '1', currency: 'USD', userId: 1, retailerId: 1 },
        { id: 2, amount: '2', currency: 'GBP', userId: 2, retailerId: 2 },
      ],
      users: {
        '1': { name: 'Joe Bloggs' },
        '2': { name: 'Customer 2' },
      },
      retailers: {
        '1': { name: 'Test Company', city: 'Test City' },
        '2': { name: 'Test Company2', city: 'Test City2' },
      },
    };

    const { container } = renderWithProviderStore(<App />, { bankStore });
    expect(container).toMatchSnapshot();
  });
});
