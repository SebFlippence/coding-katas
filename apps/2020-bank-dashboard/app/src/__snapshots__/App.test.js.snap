// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders loading screen 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      Transactions
    </h1>
    <p>
      Loading...
    </p>
  </div>
</div>
`;

exports[`App renders with transaction & user data, with retailer data loading 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      Transactions
    </h1>
    <table>
      <thead>
        <tr>
          <th>
            Amount
          </th>
          <th>
            User
          </th>
          <th>
            Retailer
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
             
            USD
          </td>
          <td>
            Joe Bloggs
          </td>
          <td>
            Loading...
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`App renders with transaction data with other data loading 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      Transactions
    </h1>
    <table>
      <thead>
        <tr>
          <th>
            Amount
          </th>
          <th>
            User
          </th>
          <th>
            Retailer
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
             
            USD
          </td>
          <td>
            Loading...
          </td>
          <td>
            Loading...
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`App renders with transaction, user & retailer data 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      Transactions
    </h1>
    <table>
      <thead>
        <tr>
          <th>
            Amount
          </th>
          <th>
            User
          </th>
          <th>
            Retailer
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
             
            USD
          </td>
          <td>
            Joe Bloggs
          </td>
          <td>
            Test Company Test City
          </td>
        </tr>
        <tr>
          <td>
            2
             
            GBP
          </td>
          <td>
            Customer 2
          </td>
          <td>
            Test Company2 Test City2
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
